### ISSUE TITLE
Checkbox in uncontrolled mode never fires onChange

### ISSUE THREAD

**Issue by zwx00** [palantir/blueprint#2405]
### Bug report

- __Package version(s)__: 

```

    "@blueprintjs/core": "^2.1.1",
    "react": "^16.3.2",
    "react-dom": "^16.3.2",
    "react-scripts": "1.1.4",
    "react-transition-group": "^2.3.1"
```

- __Browser and OS versions__: 

FF 59.0 on linux, chromium on linux

#### Steps to reproduce
Install blueprintjs/core on blank create-react-app.

index.js:
```

import React, { Component } from "react";
import ReactDOM from "react-dom";
import { Checkbox } from "@blueprintjs/core";


ReactDOM.render(<Checkbox onChange={(input) => { console.log("lala"); }} label="Show ooonly &quot;On Hold&quot;" />

```
I also tried other variations, such as:

```
class CheckboxControlled extends Component {
	constructor(props) {
		super(props);
		this.state = {
			value: false
		};

		this.changeValue = this.changeValue.bind(this);
	}

	changeValue(value) {
		console.log(value);
		this.setState({ value: !this.state.value });
	}

	render() {
		return <Checkbox checked={this.state.value} onChange={this.changeValue} label="Show ooonly &quot;On Hold&quot;" />;
	}
}

ReactDOM.render(<CheckboxControlled />, document.getElementById("root"));


```

#### Actual behavior
Clicking on the checkbox should fire the function. The function I pass to onChange prop is never called. 

#### Expected behavior

According to the documentation, I'd expect onChange to work much like it does with other input elements, eg. when user inputs some information (clicks on the checkbox), onChange is called, I update its value externally and the component updates itself accordingly.

**Comment by clekili** [comment]
Seems like this issue was addressed at #2370 
@giladgray to confirm

**Comment by giladgray** [comment]
yes fixed already. will ship today.
